test/explicit-resource-management/main.test.ts

```diff
@@ -0,0 +1,120 @@
+
+import { expect } from 'chai';
+import { describe, it } from 'mocha';
+import { AbstractCursor, ChangeStream, ClientSession, GridFSBucket, MongoClient } from 'mongodb/lib/beta';
+import * as sinon from 'sinon';
+import { Readable } from 'stream';
+import { pipeline } from 'stream/promises';
+import { setTimeout } from 'timers/promises';
+
+async function setUpCollection(client: MongoClient) {
+	const collection = client.db('foo').collection<{ name: string }>('bar');
+	const documents: Array<{ name: string }> = Array.from({ length: 5 }).map(i => ({
+		name: String(i)
+	}));
+	await collection.insertMany(documents)
+	return collection;
+}
+
+describe('explicit resource management smoke tests', function () {
+	const clientSpy = sinon.spy(MongoClient.prototype, Symbol.asyncDispose);
+	const cursorSpy = sinon.spy(AbstractCursor.prototype, Symbol.asyncDispose);
+	const endSessionSpy = sinon.spy(ClientSession.prototype, Symbol.asyncDispose);
+	const changeStreamSpy = sinon.spy(ChangeStream.prototype, Symbol.asyncDispose);
+	const readableSpy = sinon.spy(Readable.prototype, Symbol.asyncDispose);
+
+	afterEach(function () {
+		clientSpy.resetHistory();
+		cursorSpy.resetHistory();
+		endSessionSpy.resetHistory();
+		changeStreamSpy.resetHistory();
+		readableSpy.resetHistory();
+	});
+
+	describe('MongoClient', function () {
+		it('can be used with await-using syntax', async function () {
+			{
+				await using client = new MongoClient(process.env.MONGODB_URI!);
+				await client.connect();
+			}
+			expect(clientSpy.called).to.be.true;
+			expect(clientSpy.callCount).to.equal(1);
+		})
+	})
+
+	describe('Cursors', function () {
+		it('can be used with await-using syntax', async function () {
+			{
+				await using client = new MongoClient(process.env.MONGODB_URI!);
+				await client.connect();
+
+				const collection = await setUpCollection(client);
+
+				await using cursor = collection.find();
+				await cursor.next();
+				await cursor.next();
+				await cursor.next();
```

Comment:
what's special about the 3 calls to next? should there also be a test where the cursor isn't iterated at all, and perhaps, a test where the cursor is fully exhausted, too?

Reply:
No reason.  I removed 2 calls.